Task: Spring Boot Portfolio Project — Part 3. Executor: Oleksii Bashkirov.

OleksiiBashkirov opened the pull request:
feat: Integrate Liquibase, implement JPA repository and soft delete
- Replaced BookRepository with JpaRepository implementation.
- Added Liquibase support for database versioning.
- Changed spring.jpa.hibernate.ddl-auto property value to validate.
- Implemented soft delete functionality in the Book entity.
- Added missing endpoints and methods in BookService.
- Updated BookController to handle new endpoints:
  - GET /api/books
  - GET /api/books/{id}
  - POST /api/books
  - PUT /api/books/{id}
  - DELETE /api/books/{id}
- Updated database schema to use BIT(1) for boolean values.
- Ensured proper integration and testing with Postman.

OleksiiBashkirov committed:
feat: Integrate Liquibase, implement JPA repository and soft delete

- Replaced BookRepository with JpaRepository implementation.
- Added Liquibase support for database versioning.
- Changed spring.jpa.hibernate.ddl-auto property value to validate.
- Implemented soft delete functionality in the Book entity.
- Added missing endpoints and methods in BookService.
- Updated BookController to handle new endpoints:
  - GET /api/books
  - GET /api/books/{id}
  - POST /api/books
  - PUT /api/books/{id}
  - DELETE /api/books/{id}
- Updated database schema to use BIT(1) for boolean values.
- Ensured proper integration and testing with Postman.

Elena-Bruyako commented on src/main/java/mate/academy/model/Book.java:
```diff
@@ -10,11 +10,15 @@
 @Getter
 @Setter
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted = false")
```
`@Where` is deprecated, use `@SQLRestriction`

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
-                        () -> new EntityNotFoundException("Book not found by id: " + id)
-                );
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
```
```suggestion
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
+                .orElseThrow(() -> new EntityNotFoundException("Book not found by id: " + id));
+        book.setTitle(requestDto.getTitle());
```
use MapStruct for this logic

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
+
+    @Override
+    public void deleteById(Long id) {
+        Book book = bookRepository.findById(id)
+                .filter(b -> !b.isDeleted())
+                .orElseThrow(() -> new EntityNotFoundException("Book not found by id: " + id));
+        book.setDeleted(true);
+        bookRepository.save(book);
```
```suggestion
        bookRepository.deleteById(id);
```

Elena-Bruyako commented on src/main/java/mate/academy/service/impl/BookServiceImpl.java:
```diff
@@ -26,15 +26,39 @@ public BookDto save(CreateBookRequestDto requestDto) {
     @Override
     public List<BookDto> findAll() {
         return bookRepository.findAll().stream()
+                .filter(book -> !book.isDeleted())
```
```suggestion
```

Elena-Bruyako reviewed (changes requested):
Let's improve your solution

OleksiiBashkirov committed:
- Refactored BookServiceImpl to use MapStruct for updating book entities.
- Removed redundant filters in the BookServiceImpl class.
- Used @SQLRestriction instead of the deprecated @Where in the Book class.
